src/components/App.tsx

```diff
@@ -1756,6 +1774,14 @@ class App extends React.Component<AppProps, AppState> {
       }),
       onSubmit: withBatchedUpdates(({ text, viaKeyboard }) => {
         const isDeleted = !text.trim();
+        if (this.props.onBeforeTextSubmit) {
+          const updatedText = this.props.onBeforeTextSubmit(
+            element,
+            text,
+            isDeleted,
+          );
+          text = updatedText ? updatedText : text;
```

Comment:
```suggestion
          text = updatedText ?? text;
```

Simplified and changed to null-coalescing — since you're required to return a string I think we should allow `""`.

But we should move this statement above the `isDeleted` check on `1777` so that returning `""` deletes the element as it should.